File: demo/src/main/java/com/example/demo/entity/Evenement.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "evenements")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Evenement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titre;

    @Column(columnDefinition = "TEXT")
    private String description;

    private Instant date;

    private String type;

    /** Lieu de rencontre / localisation de l'événement (adresse ou nom du lieu). */
    @Column(name = "lieu")
    private String lieu;

    /** Latitude (optionnel, pour affichage carte). */
    private Double latitude;

    /** Longitude (optionnel, pour affichage carte). */
    private Double longitude;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "groupe_id", nullable = false)
    private GroupeRunning groupe;

    @OneToOne(mappedBy = "evenement", cascade = CascadeType.ALL, orphanRemoval = true)
    private Historique historique;

    /** Rappel « 1h avant » déjà envoyé (évite doublons). */
    @Column(name = "rappel_1h_envoye", nullable = false, columnDefinition = "boolean not null default false")
    private boolean rappel1hEnvoye = false;

    /** Rappel « 24h avant » déjà envoyé (évite doublons). */
    @Column(name = "rappel_24h_envoye", nullable = false, columnDefinition = "boolean not null default false")
    private boolean rappel24hEnvoye = false;
}

File: app/lib/models/evenement.dart
import 'groupe_running.dart';

class Evenement {
  final int id;
  final String titre;
  final String? description;
  final DateTime? date;
  final String? type;
  /// Lieu de rencontre / localisation de l'événement.
  final String? lieu;
  final double? latitude;
  final double? longitude;
  final GroupeRunning? groupe;

  Evenement({
    required this.id,
    required this.titre,
    this.description,
    this.date,
    this.type,
    this.lieu,
    this.latitude,
    this.longitude,
    this.groupe,
  });

  factory Evenement.fromJson(Map<String, dynamic> json) {
    return Evenement(
      id: json['id'] as int,
      titre: json['titre'] as String,
      description: json['description'] as String?,
      date: json['date'] != null
          ? DateTime.tryParse(json['date'].toString())
          : null,
      type: json['type'] as String?,
      lieu: json['lieu'] as String?,
      latitude: json['latitude'] != null
          ? (json['latitude'] is int
              ? (json['latitude'] as int).toDouble()
              : json['latitude'] as double?)
          : null,
      longitude: json['longitude'] != null
          ? (json['longitude'] is int
              ? (json['longitude'] as int).toDouble()
              : json['longitude'] as double?)
          : null,
      groupe: json['groupe'] != null
          ? GroupeRunning.fromJson(json['groupe'] as Map<String, dynamic>)
          : null,
    );
  }
}
